core/src/main/java/com/amazon/opendistroforelasticsearch/sql/expression/aggregation/MinAggregator.java

```diff
@@ -0,0 +1,106 @@
+import static com.amazon.opendistroforelasticsearch.sql.data.model.ExprValueUtils.integerValue;
+import static com.amazon.opendistroforelasticsearch.sql.data.model.ExprValueUtils.longValue;
+import static com.amazon.opendistroforelasticsearch.sql.utils.ExpressionUtils.format;
+
+import com.amazon.opendistroforelasticsearch.sql.data.model.ExprNullValue;
+import com.amazon.opendistroforelasticsearch.sql.data.model.ExprType;
+import com.amazon.opendistroforelasticsearch.sql.data.model.ExprValue;
+import com.amazon.opendistroforelasticsearch.sql.exception.ExpressionEvaluationException;
+import com.amazon.opendistroforelasticsearch.sql.expression.Expression;
+import com.amazon.opendistroforelasticsearch.sql.expression.function.BuiltinFunctionName;
+import com.amazon.opendistroforelasticsearch.sql.storage.bindingtuple.BindingTuple;
+import java.util.List;
+
+/**
+ * The minimum aggregator aggregate the value evaluated by the expression.
+ * If the expression evaluated result is NULL or MISSING, then the result is NULL.
+ */
+public class MinAggregator extends Aggregator<MinAggregator.MinState> {
+
+  public MinAggregator(List<Expression> arguments, ExprType returnType) {
+    super(BuiltinFunctionName.MIN.getName(), arguments, returnType);
+  }
+
+
+  @Override
+  public MinState create() {
+    return new MinState(returnType);
+  }
+
+  @Override
+  public MinState iterate(BindingTuple tuple, MinState state) {
+    Expression expression = getArguments().get(0);
+    ExprValue value = expression.valueOf(tuple);
+    if (value.isNull() || value.isMissing()) {
+      state.isNullResult = true;
+    } else {
+      state.min(value);
+    }
+    return state;
+  }
+
+  @Override
+  public String toString() {
+    return String.format("min(%s)", format(getArguments()));
+  }
+
+  protected class MinState implements AggregationState {
+    private final ExprType type;
+    private ExprValue minResult;
+    private boolean isNullResult;
+
+    public MinState(ExprType type) {
+      this.type = type;
+      minResult = doubleValue(Double.MAX_VALUE);
+      isNullResult = false;
+    }
+
+    public void min(ExprValue value) {
+      switch (type) {
+        case INTEGER:
+          minResult = integerValue(Math.min(getIntegerValue(minResult), getIntegerValue(value)));
+          break;
+        case LONG:
+          minResult = longValue(Math.min(getLongValue(minResult), getLongValue(value)));
+          break;
+        case FLOAT:
+          minResult = floatValue(Math.min(getFloatValue(minResult), getFloatValue(value)));
+          break;
+        case DOUBLE:
+          minResult = doubleValue(Math.min(getDoubleValue(minResult), getDoubleValue(value)));
+          break;
+        default:
+          throw new ExpressionEvaluationException(
+              String.format("unexpected type [%s] in min aggregation", type));
+      }
+    }
+
+    @Override
+    public ExprValue result() {
+      return isNullResult ? ExprNullValue.of() : minResult;
```

Comment:
So if any field is `NULL`, then null is returned as result? Are we following some spec here? I checked MySQL and ANSI SQL, it seems "All aggregate functions except COUNT(*) will ignore NULL values when computing their results." Unless all fields are null, result won't be null eventually.

For example, the result is 2 for the following test data and query.
```
CREATE TABLE Test(age INT);

INSERT INTO Test VALUES(1);
INSERT INTO Test VALUES(null);
INSERT INTO Test VALUES(2);

SELECT MAX(age) FROM Test
```

Reply:
+1, we should follow https://asterixdb.apache.org/docs/0.9.3/sqlpp/manual.html#Aggregation_functions ARRAY_XXXX
Our previous implementation of SUM, AVG, COUNT also need to be changed.